### PR TITLE
Issue #6 - Image Tags

### LINKED ISSUE #6: Image Tags
#### As a user I want to add tags to images.

__Story__:

Our users are loving the ability to add images to our service. However, now
that there are more than a handful of images, it is apparent that we need some
way to filter images by certain categories. In order to do that, we first need
to label images in some way. It is time to add the ability for our users to tag
images when they add them.

__Acceptance criteria__:
- [x] When I add a new image, there is a form field to add tags.
- [x] After I've saved an image with tags, I see those tags displayed with the
  associated image.
- [x] My application uses the acts-as-taggable-on gem to provide the tagging
  functionality.

__Reference material__:
- https://github.com/mbleigh/acts-as-taggable-on

__Note__: You should allow images to have no tags; this sets the stage for a later issue.

__Dependencies__:
- Image Index

### PR DESCRIPTION
**Pull request by cvetanovskaa**
Closes #6

---

The following is a checklist to remind you of the steps to take when opening a
new PR. The horizontal line above and everything below can be removed from the
pull request description.

- [X] I have replaced `X` above with the associated issue number.
- [X] I have changed the issue's label to "pull request".
- [X] I have assigned the designated reviewer to this PR.
- [X] I have set the appropriate label on this PR.
- [X] I have written clear commit messages ([ref](http://chris.beams.io/posts/git-commit/)).

If needed, please review the document on the
[Image Sharing Project](https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project)
for information on how labels should be used.

### CODE REVIEW

**Comment by cvetanovskaa** [comment]
[Ropes Issue#6: Image Tags](https://trello.com/c/GqERqkZh/537-ropes-issue6-image-tags)

File: app/views/images/index.html.erb [hunk 1]
```diff
@@ -13,7 +13,13 @@
     <div class="row">
       <% @images.each do |img| %>
         <div class="image_card--card col-md-4 mt-2">
+
           <%= image_tag(img.url, :class => 'image_format card-img-top') %>
+          <div class="pt-2 pb-2 ">
+            <% img.tag_list.each do |tag| %>
+              <%= link_to tag, :class => 'card-link' %>
```
[the remark this answers is not in the document]
Comment (cvetanovskaa) [reply app/views/images/index.html.erb]: Changed all :class =>, to class:

File: app/views/images/show.html.erb [hunk 2]
```diff
@@ -5,11 +5,18 @@
   </div>
 <% end %>
 
-<div class="container">
+<div class="container ">
```
[the remark this answers is not in the document]
Comment (cvetanovskaa) [reply app/views/images/show.html.erb]: Woops. Got it.

File: db/migrate/20190118235428_create_images.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class CreateImages < ActiveRecord::Migration[5.2]
```
Comment (CRiva) [inline db/migrate/20190118235428_create_images.rb]: I think this should be it's own commit, rather than in this grouping, cause its different than the commit message suggests is going on.

File: app/models/image.rb [hunk 4]
```diff
@@ -1,4 +1,5 @@
 class Image < ApplicationRecord
+  acts_as_taggable_on :tags
```
Comment (CRiva) [inline app/models/image.rb]: can we get a test written for this?

File: db/migrate/20190118235428_create_images.rb (hunk 3 above)
Comment (cvetanovskaa) [reply db/migrate/20190118235428_create_images.rb]: split the last commit in 2

File: app/models/image.rb (hunk 4 above)
Comment (cvetanovskaa) [reply app/models/image.rb]: added test
